http: metadata, content-disposition

hiddenmarten opened the pull request:
<!--
Thank you very much for contributing code or documentation to rclone! Please
fill out the following questions to make it easier for us to review your
changes.

You do not need to check all the boxes below all at once, feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box.
-->

#### What is the purpose of this change?

The objective is to extract the file name from the content-disposition header. This name should replace the one in stat, and a content-disposition handler should be provided via serve.

#### Was the change discussed in an issue or in the forum before?

https://github.com/rclone/rclone/issues/8298

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [X] I have added tests for all changes in this PR if appropriate.
- [X] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

hiddenmarten commented:
Hi @ncw!

May I get a review, please?

hiddenmarten commented:
Bump @ncw .
:)

ncw reviewed (commented):
Sorry for the delay reviewing.

Adding metadata to the HTTP backend seems like a good idea, though if we were starting from that then there are lot of other things we could add other than Content-Disposition.

I don't like the changes in lsjson/operations - any parsing should be done in the HTTP backend. So set another metadata key, eg `content-disposition-filename`. I don't think any changes should be needed to lsjson/operations.

The http serve changes are interesting, but again maybe should support more metadata than just content-disposition.

A minimum set of metadata for the http backend and server might be something like

- cache-control
- content-disposition
- content-encoding
- content-language
- content-type

What do you think?

hiddenmarten commented:
Just pulled an actual master branch and rebased on it, sorry.

hiddenmarten commented:
Hi @ncw,

The changes are ready for review. Here’s what’s been done:

- Updated field names to match the S3 backend. (This change can be reverted if necessary.)
- Added system metadata fields, as per your suggestion.
- Reverted the modifications in `lsjson/operations` and updated the `Remote()` method to use the `contentDispositionFilename` when available.

Please let me know if you need any further improvements or changes.

hiddenmarten commented:
@ncw ping :)

hiddenmarten commented:
@ncw ping :)
